### DISCUSSION TITLE
Prev/Next devlopment preset

### DISCUSSION THREAD

**Discussion by vincesd8** [rsjaffe/MIDI2LR#579] [General]
Would it be possible to add an action of prev/next preset, where I can bind it to a nob?

**Comment by vincesd8** [comment]
As there are upto 80 dev preset possible, being able to go thru them with one knob is much more convinent than mapping each to a button.

**Reply by rsjaffe** [reply]
I'll look into this. The main problem is that the number of assigned presets will vary from user to user, and the program will have to figure out how to manage that.

**Comment by rsjaffe** [comment]
FIgured it out! If you want to try it out, download [plugin.zip](https://github.com/rsjaffe/MIDI2LR/files/7707480/plugin.zip), extract the files from the zip, and move them into the MIDI2LR.lrplugin directory. Then, before starting Lightroom, delete, the _MenuTrans.xml_ file (this forces MIDI2LR to regenerate the menu). The file is located in _%AppData%\MIDI2LR_ (Windows users) or _~/Library/Application Support/MIDI2LR_ (macOS users).

**Comment by vincesd8** [comment]
Thank you so much! I tested it out. It is working well general! But sometimes I get an error pop up: 

"An internal error has occurred: the catalog could not be updated with additional module metadata.PastePreset." I was able to continue after clicking ok to dismiss the pop-up.

This seem to happen after I randomly twist the knob back and forth for 10~20 sec.

Another enhancement would be ability to do prev/next within a sub-range. For example, one knob to select between all variants of "wedding" profile (say profile 1~10), while another knob to switch between all "fashion" profiles, say profile (11~20).

**Reply by rsjaffe** [reply]
Sub-range is too complicated to produce. I suggest having separate profiles that have just the sub-ranges of presets you want to test, and load that profile when you want to select variants of a given profile.

**Comment by rsjaffe** [comment]
Yeah, that error is because Lightroom can't keep up with your knob twisting! I've added some rate-limiting that should fix the problem. Please test it out. The attached zip file has one file to place inside your MIDI2LR.lrplugin directory. No other changes needed.
[Presets.zip](https://github.com/rsjaffe/MIDI2LR/files/7713647/Presets.zip)

**Reply by vincesd8** [reply]
I see. Appreciate the help. This is already very helpful.

**Comment by vincesd8** [comment]
But dev profile seems to be a global array set fro light room general
options and not per profile? Am I missing something?

**Reply by rsjaffe** [reply]
No, you’re not. My mistake. I don’t have a good answer for your idea, then.
